include/zephyr/drivers/rtcc.h

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) 2022 Trackunit Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file drivers/rtcc.h
+ * @brief Public real time clock calendar driver API
+ */
+
+#ifndef ZEPHYR_INCLUDE_RTCC_H_
+#define ZEPHYR_INCLUDE_RTCC_H_
+
+/**
+ * @brief RTCC Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+#include <errno.h>
+#include <time.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief RTCC alarm triggered handler
+ * @note This handler is likely to be invoked from within an ISR context
+ * @param dev Device instance invoking the handler
+ * @param id Alarm id
+ * @param user_data Optional user_data provided with the alarm configuration
+ */
+typedef void (*rtcc_alarm_triggered_handler_t)(const struct device *dev, size_t id,
+					       void *user_data);
+
+/**
+ * @name RTCC alarm trigger mask
+ * @brief Mask for tm fields to trigger alarm
+ * @{
+ */
+#define RTCC_ALARM_MATCH_SECOND          BIT(0)
+#define RTCC_ALARM_MATCH_MINUTE          BIT(1)
+#define RTCC_ALARM_MATCH_HOUR            BIT(2)
+#define RTCC_ALARM_MATCH_DAY             BIT(3)
+#define RTCC_ALARM_MATCH_MONTH           BIT(4)
+#define RTCC_ALARM_MATCH_YEAR            BIT(5)
+/**
+ * @}
+ */
+
+/**
+ * @typedef rtcc_alarm_trigger_mask
+ * @brief Type storing bitmask of RTCC alarm triggers
+ */
+typedef uint8_t rtcc_alarm_match_mask_t;
+
+/**
+ * @brief RTCC alarm configuration
+ * @param time Date and time to match
+ * @param mask Parameters of the date and time to match
+ * @param handler Handler invoked once the alarm is triggered
+ * @param user_data Optional user data which passed to the handler when invoked
+ * @note Handler and user data will be lost of device resets before alarm is triggered
```

Comment:
missing documentation end, does it compile?

Reply:
It does not :) It is a draft to discuss which functions should be in which API